master/internal/experiment.go

```diff
@@ -284,8 +284,15 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 
 	// Experiment shutdown logic.
 	case actor.PostStop:
-		if err := e.db.SaveExperimentProgress(e.ID, nil); err != nil {
-			ctx.Log().Error(err)
+		if e.State == model.CompletedState || e.State == model.StoppingCompletedState {
+			setProgress := 1.0
+			if err := e.db.SaveExperimentProgress(e.ID, &setProgress); err != nil {
+				ctx.Log().Error(err)
+			}
+		} else {
+			if err := e.db.SaveExperimentProgress(e.ID, nil); err != nil {
```

Comment:
why not just do nothing here? i think i mentioned this, it seems really useful to see an experiment failed at 38% not nil. if the UI doesn't want to expose that, it's fine to me, but I see no reason to erase useful info.

Reply:
That does make sense. In my latest commit I remove this call and update the release note.